### PR TITLE
Compare value length to used length () in reading data from message buffer.

### PR DESCRIPTION
**Pull request by Hadatko**
Signed-off-by: Cervenka Dusan <cervenka@acrios.com>

# Pull request

**Choose Correct**

- [X] bug
- [ ] feature

**Describe the pull request**

This should cover cases when we will try read more data that we received.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

- OS:
- eRPC Version:

**Steps you didn't forgot to do**

- [X] I checked if other PR isn't solving this issue.
- [X] I read Contribution details and did appropriate actions.
- [X] PR code is tested.
- [X] PR code is formatted.

**Additional context**

### CODE REVIEW

**Comment by amgross** [comment]
I added my comments, basically it looks great, thank you very much for the hard work

**Comment by amgross** [comment]
Does it planned to release at 1.9.1? (same question for #287 and #276 ), I see in the milestones that release 1.9.1 has delay.

**Comment by Hadatko** [comment]
I don't see any of your comments in this PR. Did you send them? 
Milestone release date is aligned with NXP sdk release which i was guessing.

**Comment by amgross** [comment]
erpc_message_buffer.cpp line 130
`erpc_assert((uint16_t)(m_pos - m_buffer->get()) + n <= m_buffer->getLength());`
I prefer here casting to uint32_t instead of 16_t, that way the '+n' can't wrapparound

erpc_message_buffer.cpp line 139
`erpc_assert((m_pos - n) >= m_buffer->get());`
Don't want to check the wraparound?

**Review by MichalPrincNXP — APPROVED** [review]

File: erpc_c/infra/erpc_message_buffer.cpp [hunk 1]
```diff
@@ -102,7 +109,56 @@ void MessageBuffer::Cursor::set(MessageBuffer *buffer)
...
+const uint8_t &MessageBuffer::Cursor::operator[](int index) const
+{
+    erpc_assert(((m_pos + index) >= m_buffer->get()) && ((uint16_t)(m_pos - m_buffer->get()) + index <= m_buffer->getLength()));
+
+    return m_pos[index];
+}
+
+MessageBuffer::Cursor &MessageBuffer::Cursor::operator+=(uint16_t n)
+{
+    erpc_assert((uint16_t)(m_pos - m_buffer->get()) + n <= m_buffer->getLength());
```
[the remark this answers is not in the document]
Comment (MichalPrincNXP) [reply erpc_c/infra/erpc_message_buffer.cpp:130]: I think uint16_t should be used to use the same type in the condition. Any thoughts, @Hadatko ?

File: erpc_c/infra/erpc_message_buffer.cpp [hunk 2]
```diff
@@ -102,7 +109,56 @@ void MessageBuffer::Cursor::set(MessageBuffer *buffer)
...
+
+    m_pos += n;
+
+    return *this;
+}
+
+MessageBuffer::Cursor &MessageBuffer::Cursor::operator-=(uint16_t n)
+{
+    erpc_assert((m_pos - n) >= m_buffer->get());
```
[the remark this answers is not in the document]
Comment (MichalPrincNXP) [reply erpc_c/infra/erpc_message_buffer.cpp:139]: Can you comment, @Hadatko ? Thanks.

File: erpc_c/infra/erpc_message_buffer.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hadatko) [reply erpc_c/infra/erpc_message_buffer.cpp:130]: I think we need use uint32_t type here, or we need another assert check if we fit into boundries.

**Comment by Hadatko** [comment]
@amgross Is it ok now?

**Review by amgross — APPROVED** [review]

**Review by MichalPrincNXP — APPROVED** [review]

**Review by MichalPrincNXP — APPROVED** [review]

**Review by MichalPrincNXP — APPROVED** [review]

**Comment by MichalPrincNXP** [comment]
Thanks for the effort!

**Pull request merged**